File: src/feature-flag-service/src/main/java/com/dynatrace/easytrade/featureflagservice/Flag.java
package com.dynatrace.easytrade.featureflagservice;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class Flag {
	private String id;
	private Boolean enabled;
	private String name;
	private String description;
	private Boolean isModifiable;
	private String tag;

	public void setEnabled(Boolean enabled) {
		if (Boolean.TRUE.equals(isModifiable)) {
			this.enabled = enabled;
		} else {
			throw new UnsupportedOperationException("Can't update a non-modifiable flag");
		}
	}
}

File: src/broker-service/BrokerService/src/ProblemPatterns/OpenFeature/Providers/FeatureFlagService/Flag.cs
namespace EasyTrade.BrokerService.ProblemPatterns.OpenFeature.Providers.FeatureFlagService;

public class Flag
{
    public string Id {get; set;}
    public bool Enabled {get; set;}
    public string Name {get; set;}
    public string Description {get; set;}
    public bool IsModifiable {get; set;}
    public string Tag {get; set;}

    public Flag(string id, bool enabled, string name, string description, bool isModifiable, string tag)
    {
        Id = id;
        Enabled = enabled;
        Name = name;
        Description = description;
        IsModifiable = isModifiable;
        Tag = tag;
    }
}